source/simulator/assembly.cc

```diff
@@ -1438,7 +1463,7 @@ namespace aspect
               (*scratch.face_finite_element_values)[introspection.extractors.velocities].get_function_values(free_surface->mesh_velocity,
                   scratch.face_mesh_velocity_values);
 
-            if (assemblers->advection_system_assembler_on_face_properties.need_face_material_model_data)
+            if (assemblers->advection_system_assembler_on_face_properties[advection_field_index].need_face_material_model_data)
```

Comment:
can we set the values to something invalid if this is false, so it will crash if someone forgets to set the flag?